alternator/ttl.hh

```diff
@@ -38,13 +39,25 @@ namespace alternator {
 class expiration_service final : public seastar::peering_sharded_service<expiration_service> {
     database& _db;
     service::storage_proxy& _proxy;
+    // _end is set by start(), and resolves when the the background service
+    // started by it ends. To ask the background service to end, _abort_source
+    // should be triggered. shutdown() below uses both _abort_source and _end.
+    std::optional<future<>> _end;
+    abort_source _abort_source;
+    bool shutting_down() { return _abort_source.abort_requested(); }
 public:
     // sharded_service<expiration_service>::start() creates this object on
     // all shards, so calls this constructor on each shard. Later, the
     // additional start() function should be invoked on all shards.
     expiration_service(database&, service::storage_proxy&);
     future<> start();
     future<> run();
+    // Call shutdown() to ask the expiration_service to shut down as quickly as
+    // it can. The returned future indicates when the service is no longer
+    // running.
+    // shutdown() may be called even before start(), but may only be called once -
+    // calling it twice will result in an exception.
+    future<> shutdown();
```

Comment:
Shouldn't it be called from stop()?

Reply:
Wow, I didn't realize that sharded_service has magic code in the case the class has a method called stop(), it calls it in each shard. Instead, I did this manually in main.cc:
```
                       es.invoke_on_all(&alternator::expiration_service::shutdown).get();
                       es.stop().get();
```
There is some benefit of renaming my shutdown() stop() and making the code one line shorter (no need for the explicit invoke_on_all). On the other hand, I have to be honest - I don't like the way that sharded_service has methods call start() and stop() and also call the object's *different* start() and stop() methods. It is very confusing.

Anyway, I'll change to stop() to conform with what the rest of the code does.

Reply:
Done in next version - renamed shutdown() to stop(), and then it gets automatically called from sharded_service::stop(). Added a comment to explain this, because it's not obvious.